op-program/host/prefetcher/prefetcher.go

```diff
@@ -312,6 +316,12 @@ func (p *Prefetcher) prefetch(ctx context.Context, hint string) error {
 			return fmt.Errorf("failed to re-execute block: %w", err)
 		}
 		return p.kvStore.Put(BlockDataKey(blockHash).Key(), []byte{1})
+	case l2.HintAgreedPrestate:
+		if len(p.agreedPrestate) == 0 {
+			return ErrAgreedPrestateUnavailable
+		}
+		hash := crypto.Keccak256Hash(p.agreedPrestate)
+		return p.kvStore.Put(preimage.Keccak256Key(hash).PreimageKey(), p.agreedPrestate)
```

Comment:
remind me again why we don't immediately write the prestate to kv when creating the prefetcher?

Reply:
@clabby didn't like it.  I think it makes sense to just follow the usual pattern and then we don't wind up rewriting it unnecessarily.